src/sentry/api/serializers/models/notification_action.py

```diff
@@ -0,0 +1,42 @@
+from typing import Dict, Sequence
+
+from sentry.api.serializers import Serializer, register
+from sentry.api.serializers.models.user import manytoone_to_dict
+from sentry.models.notificationaction import (
+    ActionService,
+    ActionTarget,
+    NotificationAction,
+    NotificationActionProject,
+)
+
+
+@register(NotificationAction)
+class NotificationActionSerializer(Serializer):
+    def get_attrs(self, item_list: Sequence[NotificationAction], user):
+        action_ids = {i.id for i in item_list}
+        projects_by_action_id = manytoone_to_dict(
+            NotificationActionProject.objects.filter(action_id__in=action_ids),
```

Comment:
is this because the same `NotificationAction` can be used for multiple projects?

Reply:
Yeah, it was a consideration built into the model since it might be helpful in the future to have actions apply across projects or have none at all :+1: